### PR TITLE
Remove not needed intermediate collection while reading DatagramPacke…

### PR DESCRIPTION
**Pull request by normanmaurer**
…ts in native transports

Motivation:

We used an intermediate collection to store the read DatagramPackets and only fired these through the pipeline once wewere done with the reading loop. This is not needed and can also increase memory usage.

Modifications:

Remove intermediate collection

Result:

Less overhead and possible less memory usage during read loop.

### CODE REVIEW

File: transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDatagramChannel.java [hunk 1]
```diff
@@ -483,7 +482,10 @@ void readReady(KQueueRecvByteAllocatorHandle allocHandle) {
                         allocHandle.lastBytesRead(remoteAddress.receivedAmount());
                         data.writerIndex(data.writerIndex() + allocHandle.lastBytesRead());
 
-                        readBuf.add(new DatagramPacket(data, (InetSocketAddress) localAddress(), remoteAddress));
+                        readPending = false;
+                        pipeline.fireChannelRead(
+                                new DatagramPacket(data, (InetSocketAddress) localAddress(), remoteAddress));
```
Comment (netkins) [inline transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDatagramChannel.java]: MAJOR Prefix this call to "localAddress" with "super.". [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2388)

File: transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannel.java [hunk 2]
```diff
@@ -514,7 +513,10 @@ void epollInReady() {
                         allocHandle.lastBytesRead(remoteAddress.receivedAmount());
                         data.writerIndex(data.writerIndex() + allocHandle.lastBytesRead());
 
-                        readBuf.add(new DatagramPacket(data, (InetSocketAddress) localAddress(), remoteAddress));
+                        readPending = false;
+                        pipeline.fireChannelRead(
+                                new DatagramPacket(data, (InetSocketAddress) localAddress(), remoteAddress));
```
Comment (netkins) [inline transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannel.java]: MAJOR Prefix this call to "localAddress" with "super.". [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2388)

Comment (normanmaurer) [inline transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannel.java]: its fine

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (c318fc7ceaecad8321374c13395c67e369467e86) and 4.0 (74d1006253)

**Pull request closed without merging**